pkg/controller/webnotifier/observer.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package webnotifier
+
+import (
+	"encoding/json"
+
+	"github.com/hyperledger/aries-framework-go/pkg/didcomm/common/service"
+)
+
+const (
+	preState  = "pre_state"
+	postState = "post_state"
+)
+
+// Notifier represents a notification dispatcher.
+type Notifier interface {
+	Notify(topic string, message []byte) error
+}
+
+// Observer instance
+type Observer struct {
+	notifier Notifier
+}
+
+// NewObserver returns observer
+func NewObserver(notifier Notifier) *Observer {
+	return &Observer{notifier: notifier}
+}
+
+// RegisterAction registers action channel to observer events
+func (o *Observer) RegisterAction(topic string, ch <-chan service.DIDCommAction) {
+	go func() {
+		for action := range ch {
+			o.notify(topic, toAction(action))
+		}
+	}()
+}
+
+// RegisterStateMsg registers state channel to observer events
+func (o *Observer) RegisterStateMsg(topic string, ch <-chan service.StateMsg) {
+	go func() {
+		for msg := range ch {
+			o.notify(topic, toStateMsg(msg))
+		}
+	}()
+}
+
+func (o *Observer) notify(topic string, v interface{}) {
+	src, err := json.Marshal(v)
+	if err != nil {
+		logger.Errorf("notify marshal: %s", err)
```

Comment:
%w for error wrapping everywhere please..

Reply:
it is logger, no need to wrap